src/api/student.js

```diff
@@ -25,17 +25,16 @@ app.post('/register', (req, res) => {
         res.failureJson(e.message);
         return;
     }
-    let salt = createToken();
     let confirmStudentEmailToken = createToken();
     let confirmBackupEmailToken = createToken();
-    let hashedPassword = saltAndHash(password, salt);
-    publish('NEW_STUDENT', {
-        studentEmail,
-        backupEmail,
-        confirmStudentEmailToken,
-        confirmBackupEmailToken,
-        hashedPassword,
-        salt
```

Comment:
The salt needs to be stored somewhere, so it can be retrieved in order to recompute the hash later.